### PR TITLE
=doc add documentation section about max-open-requests problems

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolInterfaceActor.scala [hunk 1]
```diff
@@ -51,6 +51,12 @@ private class PoolInterfaceActor(gateway: PoolGateway)(implicit fm: Materializer
   private[this] val inputBuffer = Buffer[PoolRequest](hcps.setup.settings.maxOpenRequests, fm)
   private[this] var activeIdleTimeout: Option[Cancellable] = None
 
+  private[this] val PoolOverflowException = new BufferOverflowException( // stack trace cannot be prevented here because `BufferOverflowException` is final
+    s"Exceeded configured max-open-requests value of [${inputBuffer.capacity}]. This means that the request queue of this pool (${gateway.hcps}) " +
+      s"has completely filled up because the pool currently does not process requests fast enough to handle the incoming request load. " +
+      "Please retry the request later. See http://doc.akka.io/docs/akka-http/current/scala/http/client-side/pool-overflow.html for " +
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolInterfaceActor.scala]: Yes very good to link to docs

**Comment by jlprat** [comment]
We should add this also to the Java docs, right?

**Comment by jrudolph** [comment]
Yes, once reviewed, I'll copy over to the Java side.

**Comment by jrudolph** [comment]
Improved wording slightly, added pointer to check for unread response entities, and copied to Java documentation.

**Pull request merged**